src/Filter/Strings.php

```diff
@@ -105,6 +105,22 @@ public static function concat($value, string $prefix = '', string $suffix = '')
         return "{$prefix}{$value}{$suffix}";
     }
 
+    /**
+     * This filter trims and removes superfluous whitespace from the given string
+     *
+     * @param string|null $value The string to compress.
+     *
+     * @return string|null
+     */
+    public static function compress(string $value = null)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        return trim(preg_replace('/\s+/', ' ', $value));
```

Comment:
`\s` will match newlines, which is not entirely apparent from the description of the function. I would change the regex to only match horizontal whitespace:
```suggestion
        return trim(preg_replace('/\h+/', ' ', $value));
```
You could also add a parameter that controls what is compressed. For newlines, you can use `\v`, which will match all vertical whitespace.

Reply:
@jncarver I have added a flag to replace vertical whitespace. The flag is defaulted to `false`